src/palimpzest/datasources/datasources.py

```diff
@@ -161,10 +161,9 @@ def __init__(self,
 
     def getItem(self, idx: int):
         filepath = self.filepaths[idx]
-        dr = DataRecord(self.schema, scan_idx=idx)
-        dr.filename = os.path.basename(filepath)
+        pdf_filename = os.path.basename(filepath)
         with open(filepath, "rb") as f:
-            dr.contents = f.read()
```

Comment:
If I recall correctly, we were hitting some errors in some demos since we expect the "contents" field to be set in any given object. We might also fix the demos that make that assumption though.

Reply:
Demos seem to be working now, so I'll assume we are okay on this for now -- but happy to revisit if that turns out not to be the case